californium-core/src/main/java/org/eclipse/californium/core/network/stack/BaseCoapStack.java

```diff
@@ -81,19 +81,39 @@ protected final void setLayers(final Layer specificLayers[]) {
 	@Override
 	public void sendRequest(final Exchange exchange, final Request request) {
 		// delegate to top
-		top.sendRequest(exchange, request);
+		try {
+			top.sendRequest(exchange, request);
+		} catch (Exception ex) {
+			request.setSendError(ex);
+		}
 	}
 
 	@Override
 	public void sendResponse(final Exchange exchange, final Response response) {
 		// delegate to top
-		top.sendResponse(exchange, response);
+		try {
+			if (exchange.getRequest().getOptions().hasObserve()) {
+				// observe- or cancel-observe-requests may have
+				// multiple responses.
+				// when observes are finished, the last response has
+				// no longer an observe option. Therefore check the
+				// request for it.
+				exchange.retransmitResponse();
+			}
```

Comment:
I move into the "try - catch". The CoapEndpoint, for some reasons, use

```
		if (exchange.checkOwner()) {
			// send response while processing exchange.
			coapstack.sendResponse(exchange, response);
		} else {
			exchange.execute(new Runnable() {
				@Override
				public void run() {
					coapstack.sendResponse(exchange, response);
				}
			});
		}
 ```

In my very first variant, I used two try-catch, but moving it reduces that. And moving the if into the try-catch makes it somehow more complete.

Reply:
The question was more about the code itself (exchange.retransmitResponse) instead of the move. (that's why I notice that my question was totally out of topic :p)

Reply:
OK. A server-side-observe-exchange gets completed with the current response. Before sending the next notify, the Exchange must be prepared to accept that.

Reply:
Ok It's a bit  clearer 
(even If I do not really understand how we do that as the function just set the exchange as "not complete".

At first sight, I was confused because I thought that `retransmitResponse` does  "retransmit the previous response)